packaging/installer/netdata-updater.sh

```diff
@@ -650,17 +654,13 @@ update_binpkg() {
   if str_in_list "${DISTRO}" "${supported_compat_names}"; then
     DISTRO_COMPAT_NAME="${DISTRO}"
   else
-    case "${DISTRO}" in
-      opensuse-leap)
-        DISTRO_COMPAT_NAME="opensuse"
-        ;;
-      rhel)
-        DISTRO_COMPAT_NAME="centos"
-        ;;
-      *)
-        DISTRO_COMPAT_NAME="unknown"
-        ;;
-    esac
+    DISTRO_COMPAT_NAME="unknown"
+    for compat_id in ${ID_LIKE}; do
+      if str_in_list "${compat_id}" "${supported_compat_names}"; then
+        DISTRO_COMPAT_NAME="${compat_id}"
+        break
+      fi
+    done
```

Comment:
The code that you’re proposing to replace is indeed incomplete (it somehow got missed when we updated the equivalent code in the kickstart script to add support for Rocky and Alma), but was very intentionally written the way it was for consistency with the kickstart script, which in turn uses the same approach for distro detection because `ID_LIKE` is not realistically reliable as an indication of what the system supports (and we would rather fall back to another install method than install a non-functional native package).

More specifically, the `ID_LIKE` variable in `/etc/os-release` does not really guarantee any actual compatibility. Particularly relevant in this case is the fact that it does not say anything about repository compatibility. A trivial example of this would be how RHEL, CentOS, Rocky, and Alma all list `fedora` in `ID_LIKE`, but require complicated supplementary handling of repositories to get certain development packages that you can just install with the default configuration on Fedora. There are all kinds of other interesting edge cases here as well (such as the fact that a Debian derivative that uses musl instead of glibc could still legitimately list `ID_LIKE="debian"`, but would be completely ABI incompatible with any dynamically linked executables), meaning that it’s just more practical to use a curated list of mappings from the `ID` value instead.